x-pack/legacy/plugins/security/public/views/management/api_keys_grid/components/api_keys_grid_page.test.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+let mockSimulate403 = false;
+let mockSimulate500 = false;
+let mockAreApiKeysEnabled = true;
+let mockIsAdmin = true;
+
+const mock403 = () => ({ body: { statusCode: 403 } });
+const mock500 = () => ({ body: { error: 'Internal Server Error', message: '', statusCode: 500 } });
+
+jest.mock('../../../../lib/api_keys_api', () => {
+  return {
+    ApiKeysApi: {
+      async checkPrivileges() {
+        if (mockSimulate403) {
+          throw mock403();
+        }
+
+        return {
+          isAdmin: mockIsAdmin,
+          areApiKeysEnabled: mockAreApiKeysEnabled,
+        };
+      },
+      async getApiKeys() {
+        if (mockSimulate500) {
+          throw mock500();
+        }
+
+        return {
+          apiKeys: [
+            {
+              creation: 1571322182082,
+              expiration: 1571408582082,
+              id: '0QQZ2m0BO2XZwgJFuWTT',
+              invalidated: false,
+              name: 'my-api-key',
+              realm: 'reserved',
+              username: 'elastic',
+            },
+          ],
+        };
+      },
+    },
+  };
+});
+
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import { ApiKeysGridPage } from './api_keys_grid_page';
+import React from 'react';
+import { ReactWrapper } from 'enzyme';
+import { EuiCallOut } from '@elastic/eui';
+
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { SectionLoading } from '../../../../../../../../../src/plugins/es_ui_shared/public/components/section_loading';
+import { NotEnabled } from './not_enabled';
+import { PermissionDenied } from './permission_denied';
+
+const waitForRender = async (
+  wrapper: ReactWrapper<any>,
+  condition: (wrapper: ReactWrapper<any>) => boolean
+) => {
+  return new Promise((resolve, reject) => {
+    const interval = setInterval(async () => {
+      await Promise.resolve();
+      wrapper.update();
+      if (condition(wrapper)) {
+        resolve();
+      }
+    }, 10);
+
+    setTimeout(() => {
+      clearInterval(interval);
+      reject(new Error('waitForRender timeout after 2000ms'));
+    }, 2000);
+  });
+};
+
+describe('ApiKeysGridPage', () => {
+  beforeEach(() => {
+    mockSimulate403 = false;
+    mockSimulate500 = false;
+    mockAreApiKeysEnabled = true;
+    mockIsAdmin = true;
+  });
+
+  it('renders a loading state when fetching API keys', async () => {
+    const wrapper = mountWithIntl(<ApiKeysGridPage />);
+
+    expect(wrapper.find(SectionLoading)).toHaveLength(1);
```

Comment:
Would using a `data-test-subj` here let us get away from having to use `// eslint-disable-next-line @kbn/eslint/no-restricted-paths` above?